wso2is_analytics/manifests/params.pp

```diff
@@ -0,0 +1,357 @@
+
+    # system configuration data
+    $packages                 = hiera_array('packages')
+    $template_list            = hiera_array('wso2::template_list')
+    $file_list                = hiera_array('wso2::file_list', undef)
+    $patch_list               = hiera('wso2::patch_list', undef)
+    $system_file_list         = hiera_hash('wso2::system_file_list', undef)
+    $directory_list           = hiera_array('wso2::directory_list', undef)
+    $cert_list                = hiera_hash('wso2::cert_list', undef)
+    $hosts_mapping            = hiera_hash('wso2::hosts_mapping')
+
+    $master_datasources       = hiera_hash('wso2::master_datasources')
+    $registry_mounts          = hiera_hash('wso2::registry_mounts', undef)
+    $carbon_home_symlink      = hiera('wso2::carbon_home_symlink')
+    $wso2_user                = hiera('wso2::user')
+    $wso2_group               = hiera('wso2::group')
+    $maintenance_mode         = hiera('wso2::maintenance_mode')
+    $install_mode             = hiera('wso2::install_mode')
+
+    if $install_mode == 'file_repo' {
+      $remote_file_url        = hiera('remote_file_url')
+    }
+
+    $install_dir              = hiera('wso2::install_dir')
+    $pack_dir                 = hiera('wso2::pack_dir')
+    $pack_filename            = hiera('wso2::pack_filename')
+    $pack_extracted_dir       = hiera('wso2::pack_extracted_dir')
+    $hostname                 = hiera('wso2::hostname')
+    $mgt_hostname             = hiera('wso2::mgt_hostname')
+    $worker_node              = hiera('wso2::worker_node')
+    $patches_dir              = hiera('wso2::patches_dir')
+    $service_name             = hiera('wso2::service_name')
+    $service_template         = hiera('wso2::service_template')
+    $usermgt_datasource       = hiera('wso2::usermgt_datasource')
+    $local_reg_datasource     = hiera('wso2::local_reg_datasource')
+    $clustering               = hiera('wso2::clustering')
+    $dep_sync                 = hiera('wso2::dep_sync')
+    $ports                    = hiera('wso2::ports')
+    $jvm                      = hiera('wso2::jvm')
+    $sso_authentication       = hiera('wso2::sso_authentication')
+    $user_management          = hiera('wso2::user_management')
+    $enable_secure_vault      = hiera('wso2::enable_secure_vault')
+
+    $remove_file_list         = hiera_array('wso2::remove_file_list', undef)
+
+    if $enable_secure_vault {
+      $secure_vault_configs   = hiera('wso2::secure_vault_configs')
+    }
+
+    $key_stores               = hiera('wso2::key_stores')
+  } else {
+
+
+    $analytics_datasources     = {
+      wso2_analytics_fs_db      => {
+        name                => 'WSO2_ANALYTICS_FS_DB',
+        description         => 'The datasource used for analytics file system',
+        driver_class_name   => 'org.h2.Driver',
+        url                 => 'jdbc:h2:repository/database/ANALYTICS_FS_DB;AUTO_SERVER=TRUE;DB_CLOSE_ON_EXIT=FALSE;LOCK_TIMEOUT=60000',
```

Comment:
As mentioned earlier, please decide whether to go for a MySQL backend and adjust the default values accordingly.